site/public/js/forum.js

```diff
@@ -2104,3 +2104,63 @@ $(() => {
 function scrollThreadListTo(element){
   $(element).get(0).scrollIntoView({behavior: "smooth", block: "center"});
 }
+
+//Only used by the posters and only on recent posts (60 minutes since posted)
+function sendAnnouncement(thread_post_content, id){
+  $(".pin-and-email-message").attr('disabled','disabled');
+  $.ajax({
+    type: 'POST',
+    url: buildCourseUrl(['forum', 'make_announcement']),
+    data: {"id": id, 'csrf_token': window.csrfToken},
+    success: function(data){
+      if (JSON.parse(data)["status"] === "success"){ 
+        pinAnnouncement(id, 1, window.csrfToken);
```

Comment:
Yeah I see a couple places where I can use . instead, some of which which I haven't written myself. I will change the parts that I wrote, but avoid the others just in case I end up breaking something unknowingly. I am a bit confused about the CSRF part though since I see other ajax calls that pass it in forum.js but others that do not.

Reply:
I'm not sure why they use CSRF in an argument. The server.js file provides the csrfToken variable everywhere needed.